packages/portfolio-contract/src/resolver/resolver.exo.ts

```diff
@@ -0,0 +1,212 @@
+/**
+ * @file Transaction Resolver - handles cross-chain transaction confirmations
+ *
+ * This resolver is responsible for tracking pending transactions and providing
+ * promises/vows that resolve when transactions are confirmed.
+ * This is an orchestration component that can be used independently of portfolio logic.
+ */
+
+import { makeTracer, mustMatch } from '@agoric/internal';
+import type { CaipChainId, AccountId } from '@agoric/orchestration';
+import type { Zone } from '@agoric/zone';
+import { type VowTools, type VowKit, VowShape } from '@agoric/vow';
+import type { ZCF, ZCFSeat, Invitation } from '@agoric/zoe';
+import { M } from '@endo/patterns';
+import type {
+  CCTPTransactionKey,
+  CCTPSettlementOfferArgs,
+} from './types.js';
+import { ResolverOfferArgsShapes } from './types.js';
+import { q, X } from '@endo/errors';
+
+type CCTPTransactionEntry = {
+  destinationAddress: AccountId;
+  amountValue: bigint;
+  vowKit: VowKit<void>;
+};
+
+const trace = makeTracer('Resolver');
+
+const ClientFacetI = M.interface('CCTPResolverClient', {
+  registerCCTPTransaction: M.call(M.string(), M.nat()).returns(VowShape),
```

Comment:
Nit: I'm not a fan of XMLHttpRequest-style capitalization. What would you think of treating all these "CCTP" initialisms as if they were complete words?
```suggestion
  registerCctpTransaction: M.call(M.string(), M.nat()).returns(VowShape),
```